### PR TITLE
SftpUploader: add support for chmod

### PR DESCRIPTION
**Pull request by tschoonj**
and two other small fixes

### CODE REVIEW

**Review by LAShemilt — COMMENTED** [review]
On running this I have encountered two bugs:

1) if I do a directory monitor the file monitor cannot find the file I get this error: 
2020-11-17 08:58:13 rfi_file_monitor.engines.file_watchdog_engine DEBUG: Monitor found C:\Users\vpf26432\Data\test\file_1.txt for event type CREATED
Exception in thread Thread-4:
Traceback (most recent call last):
  File "C:\Users\vpf26432\Anaconda3\envs\rfi-file-monitor\lib\threading.py", line 950, in _bootstrap_inner
    self.run()
  File "C:\Users\vpf26432\Anaconda3\envs\rfi-file-monitor\lib\site-packages\watchdog\observers\api.py", line 203, in run
    self.dispatch_events(self.event_queue, self.timeout)
  File "C:\Users\vpf26432\Anaconda3\envs\rfi-file-monitor\lib\site-packages\watchdog\observers\api.py", line 376, in dispatch_events
    handler.dispatch(event)
  File "C:\Users\vpf26432\Anaconda3\envs\rfi-file-monitor\lib\site-packages\watchdog\events.py", line 452, in dispatch
    super(PatternMatchingEventHandler, self).dispatch(event)
  File "C:\Users\vpf26432\Anaconda3\envs\rfi-file-monitor\lib\site-packages\watchdog\events.py", line 331, in dispatch
    {
  File "c:\users\vpf26432\pycharmprojects\rfi-file-monitor\rfi_file_monitor\engines\file_watchdog_engine.py", line 159, in on_created
    _file = RegularFile(file_path, relative_file_path, get_file_creation_timestamp(file_path), FileStatus.CREATED)
  File "c:\users\vpf26432\pycharmprojects\rfi-file-monitor\rfi_file_monitor\utils\__init__.py", line 111, in get_file_creation_timestamp
    creation_timestamp = os.stat(file_path).st_ctime
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'C:\\Users\\vpf26432\\Data\\test\\file_1.txt'


2) If I use the temp file generator, the files get created and queued but they are not moved to sftp uploader. Just hangs.

File: setup.cfg [hunk 1]
```diff
@@ -49,7 +49,7 @@ rfi_file_monitor.engines =
     FileWatchdogEngine = rfi_file_monitor.engines.file_watchdog_engine:FileWatchdogEngine
     TemporaryFileEngine = rfi_file_monitor.engines.temporary_file_engine:TemporaryFileEngine
     AWSS3BucketEngine = rfi_file_monitor.engines.aws_s3_bucket_engine:AWSS3BucketEngine
-gui_scripts =
+console_scripts =
```
Comment (tschoonj) [inline setup.cfg:52]: This has to be undone 😄

File: rfi_file_monitor/engines/file_watchdog_engine.py [hunk 2]
```diff
@@ -156,7 +157,12 @@ def on_created(self, event):
         relative_file_path = PurePath(os.path.relpath(file_path, self._engine.params.monitored_directory))
         if self._engine.props.running and \
             self._engine._appwindow._queue_manager.props.running:
-            _file = RegularFile(file_path, relative_file_path, get_file_creation_timestamp(file_path), FileStatus.CREATED)
+            creation_timestamp = get_file_creation_timestamp(file_path)
+            if creation_timestamp:
+                _file = RegularFile(file_path, relative_file_path, creation_timestamp, FileStatus.CREATED)
+            else:
+                logger.debug("File Not found, skipped")
```
Comment (tschoonj) [inline rfi_file_monitor/engines/file_watchdog_engine.py:164]: Can you include the full file name in the debug statement? Thanks!

File: rfi_file_monitor/engines/file_watchdog_engine.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 from ..engine import Engine
 from ..utils import LongTaskWindow, get_patterns_from_string, get_file_creation_timestamp
 from ..file import FileStatus, RegularFile
-from ..utils.exceptions import AlreadyRunning, NotYetRunning
+from ..utils.exceptions import AlreadyRunning, NotYetRunning, SkippedOperation
```
Comment (tschoonj) [inline rfi_file_monitor/engines/file_watchdog_engine.py:12]: This is causing the LGTM warning, please remove SkippedOperation

**Pull request merged**